test/e2e/preflight_crdupgradesafety_valid_field_change_default_no_change_test.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2024 The Carvel Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package e2e
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+// No change in the default value of a field is a valid case
+// and this test covers that.
+func TestPreflightCRDUpgradeSafetyValidFieldChangeDefaultNoChange(t *testing.T) {
+	env := BuildEnv(t)
+	logger := Logger{}
+	kapp := Kapp{t, env.Namespace, env.KappBinaryPath, logger}
+
+	testName := "preflightcrdupgradesafetyvalidfieldchangedefaultnochange"
+
+	base := `
+---
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  annotations:
+    controller-gen.kubebuilder.io/version: v0.13.0
+  name: memcacheds.__test-name__.example.com
+spec:
+  group: __test-name__.example.com
+  names:
+    kind: Memcached
+    listKind: MemcachedList
+    plural: memcacheds
+    singular: memcached
+  scope: Namespaced
+  versions:
+  - name: v1alpha1
+    schema:
+      openAPIV3Schema:
+        properties:
+          apiVersion:
+            type: string
+          kind:
+            type: string
+          metadata:
+            type: object
+          spec:
+            type: string
+            default: foo
+          status:
+            type: object
+        type: object
+    served: true
+    storage: true
+    subresources:
+      status: {}
+`
+
+	base = strings.ReplaceAll(base, "__test-name__", testName)
+	appName := "preflight-crdupgradesafety-app"
+
+	cleanUp := func() {
+		kapp.Run([]string{"delete", "-a", appName})
+	}
+	cleanUp()
+	defer cleanUp()
+
+	update := `
```

Comment:
This "updated" CRD looks the same. Do we know for sure that this would trigger an "upgrade" operation via kapp since there is no diff (if I'm missing the diff please let me know!)? If it does _not_ trigger an upgrade operation via kapp then I'm not sure this test is really providing any value. Maybe there is another way we can update this in a way that it triggers an "upgrade"?

 Simplest one that comes to my mind is maybe updating the annotations to force a diff

Reply:
You're right, there's no diff so I don't think it will trigger an upgrade operation. I've forced the upgrade in the [latest commit](https://github.com/carvel-dev/kapp/pull/950/commits/97a258940768df26f84ba5297362f15de661d0da) via a change in annotation as you'd suggested.